### PR TITLE
Update splinterd to load signing keys from config directory

### PR DESCRIPTION
**Pull request by agunde406**
The splinterd will use these keys to execute challenge authorization.
The splinterd will load every private key stored in the config_dir/keys.
This is behind the experimental feature "challenge-authorization"

### CODE REVIEW

File: splinterd/Cargo.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ bytes = "0.4"
 clap = "2.32"
 crossbeam-channel = "0.3"
 ctrlc = "3.0"
-cylinder = "0.2"
+cylinder = { version = "0.2.2", features = ["key-load"] }
```
[the remark this answers is not in the document]
Comment (agunde406) [reply splinterd/Cargo.toml]: fixed

File: splinterd/src/daemon.rs [hunk 2]
```diff
@@ -1272,6 +1286,9 @@ impl SplinterDaemonBuilder {
             CreateError::MissingRequiredField("Missing field: strict_ref_counts".to_string())
         })?;
 
+        #[cfg(feature = "challenge-authorization")]
+        let signers = load_signer_keys(&config_dir)?;
```
[the remark this answers is not in the document]
Comment (agunde406) [reply splinterd/src/daemon.rs]: updated to be moved up a level

File: splinterd/src/daemon.rs [hunk 3]
```diff
@@ -1586,7 +1610,13 @@ pub enum CreateError {
     MissingRequiredField(String),
 }
 
-impl Error for CreateError {}
+impl Error for CreateError {
```
Comment (peterschwarz) [inline splinterd/src/daemon.rs:1613]: This change looks unnecessary

Comment (agunde406) [reply splinterd/src/daemon.rs]: Removed, it was left over after a previous change.

**Review by peterschwarz — APPROVED** [review]

**Review by Caleb-Hill — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

**Pull request merged**